### PR TITLE
Add ms build15

### PR DESCRIPTION
**Pull request by adsurg**
Hi there

I've attempted to add msbuild 15 support. I can't see how I mock the Professional / BuildTools directory choice in a test, but hopefully the change is OK.

### CODE REVIEW

File: lib/msbuild-finder.js [hunk 1]
```diff
@@ -57,13 +57,25 @@ module.exports.find = function (options) {
...
+  if (version === '12.0' || version === '14.0' || version === '15.0') {
     // On 64-bit systems msbuild is always under the x86 directory. If this
     // doesn't exist we are on a 32-bit system. See also:
     // https://blogs.msdn.microsoft.com/visualstudio/2013/07/24/msbuild-is-now-part-of-visual-studio/
     var pathRoot = env_var_dir || path.join('C:', is64Bit ? 'Program Files (x86)' : 'Program Files');
     var x64_dir = is64Bit ? 'amd64' : '';
-    return path.join(pathRoot, 'MSBuild', version, 'Bin', x64_dir, 'MSBuild.exe');
+    var msbuildRoot = pathRoot
+
+    if(version === '15.0')
```
Comment (hoffi) [inline lib/msbuild-finder.js]: Aren't these additional possible paths to the msbuild executable? If so it should fallback to the path in line 82 if both paths aren't available.

I've only found [this article](https://docs.microsoft.com/de-de/visualstudio/msbuild/what-s-new-in-msbuild-12-0), but none informations about where msbuild 15 is located, so I could be wrong.

File: test/msbuild-finder.js [hunk 2]
```diff
@@ -186,10 +220,11 @@ describe('msbuild-finder', function () {
       var expectedResult = path.join(pathRoot, 'MSBuild', expectMSBuildVersion, 'Bin', 'MSBuild.exe');
 
       mock = this.sinon.mock(fs);
```
Comment (hoffi) [inline test/msbuild-finder.js]: You need to use these mocks in your other tests, so the tests will work as expected on other systems.

Comment (adsurg) [inline test/msbuild-finder.js]: cheers, I'll sort this out.

File: lib/msbuild-finder.js (hunk 1 above)
Comment (adsurg) [inline lib/msbuild-finder.js]: I'd misunderstood this, Visual Studio defaults to install at "\Program Files (x86)\Microsoft Visual Studio\2017" with a subfolder of Enterprise, Professional, Community or BuildTools depending on which flavour(s) you've installed.  MsBuild looks for adjacent targets which are more complete in the more expensive installations. It makes sense to check all of these folders.  I'll leave it dropping back to MsBuild\15.0\Bin, but that can never exist as far as I can see.

**Comment by adsurg** [comment]
Hi there @hoffi 

Would you have a chance to take another look over this please.

Thanks in advance.

Adam

**Comment by hoffi** [comment]
Looks good, thanks! 👍

**Pull request merged**